### PR TITLE
PNG Screenshots

### PR DESCRIPTION
**Pull request by dylwhich**
### Description

This PR adds PNG screenshots using `stb_image_write`, and lets you take them with F12 instead of only with a replay script. To support that in a sane way, rawdraw is also updated for upstream changes that add defines for the key constants. And the emu extension interface is updated a bit to make working with modifier keys easier.

### Test Instructions

Run the emulator, hit F12 and take some screenshots. They should be written out to `screenshot-xxxx.png` with xxxx being a seconds+milliseconds unix timestamp. Run the emulator with `-r` and take some screenshots. Replay the recording with `-p`. The screenshot-taking should be recorded and replayed properly.

### Readiness Checklist
- [X] I have run `make format` to format the changes
- [X] I have compiled the firmware and the changes have no warnings
- [X] I have compiled the emulator and the changes have no warnings
- [X] I have run `make cppcheck` and checked that `cppcheck_result.txt` has no warnings for the changes
- [X] I have added doxygen comments to any code used by more than one Swadge mode. This includes `/*! \file` comments with Design Philosophy, Usage, and Example sections for new headers.
- [X] I have run `make docs` and checked that `doxy_warnings.txt` has no warnings for the new code

### CODE REVIEW

File: emulator/src-lib/stb_image_write.h
Comment (AEFeinstein) [inline emulator/src-lib/stb_image_write.h:1]: After having just been on the submodule warpath, I thought, wouldn't this be a nice submodule too? But https://github.com/nothings/stb/ is big and you can't submodule a single file. Oh well.

File: emulator/src/emu_main.c [hunk 1]
```diff
@@ -446,37 +445,89 @@ static void plotRoundedCorners(uint32_t* bitmapDisplay, int width, int height, i
  */
 void HandleKey(int keycode, int bDown)
 {
+    static modKey_t modifiers = EMU_MOD_NONE;
+
+// https://gcc.gnu.org/bugzilla/show_bug.cgi?id=69602
+// https://stackoverflow.com/a/26003732/882406
+// I can't help that EMU_KEY_ALT and EMU_KEY_LEFT_ALT might be identical...
```
Comment (AEFeinstein) [inline emulator/src/emu_main.c:452]: 👍

File: emulator/src/extensions/emu_ext.h [hunk 2]
```diff
@@ -94,6 +96,99 @@ typedef enum
...
+    EMU_MOD_NONE  = 0x00,
+    EMU_MOD_ALT   = 0x01,
+    EMU_MOD_CTRL  = 0x02,
+    EMU_MOD_SHIFT = 0x04,
+    EMU_MOD_SUPER = 0x08,
+} modKey_t;
+
+/**
+ * @brief Normalized key-codes across all platforms for use by emulator extensions
+ * These are just the CNFG keycodes renamed so extensions don't have to include rawdraw directly.
```
Comment (AEFeinstein) [inline emulator/src/extensions/emu_ext.h:110]: Eeh, how bad is including rawdraw directly? I look at this and see one more thing to maintain (even if keyboards aren't likely to change any time soon).

**Review by AEFeinstein — APPROVED** [review]

File: emulator/src/extensions/emu_ext.h (hunk 2 above)
Comment (AEFeinstein) [reply emulator/src/extensions/emu_ext.h:110]: @dylwhich I'm good to merge if you stand by the normalized key codes

**Comment by AEFeinstein** [comment]
Yeah, and you can have emu_ext.h include the header with the CNFG key codes
so extensions wouldn't have to add another include either.

File: emulator/src/extensions/tools/ext_tools.c [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    {
+        name = getScreenshotName(buf, sizeof(buf) - 1);
+    }
+
+    printf("Saving screenshot %s\n", name);
+
+    // Notify the replay ext of the screenshot
+    recordScreenshotTaken(name);
+
+    // Add full transparency for the rounded corners
```
[the remark this answers is not in the document]
Comment (dylwhich) [reply emulator/src/extensions/tools/ext_tools.c:166]: They might as well be ready to go straight into some sort of manual!

**Comment by dylwhich** [comment]
Alright, that should do it. Just waiting on the build now.

**Pull request merged**